llvm/tools/llvm-rc/ResourceScriptParser.cpp

```diff
@@ -419,18 +419,18 @@ RCParser::parseSingleOptionalStatement(OptStmtType StmtsType) {
   if (TypeToken->equals_insensitive("VERSION"))
     return parseVersionStmt();
 
-  if (StmtsType != OptStmtType::BasicStmt) {
```

Comment:
The condition `if (StmtsType != OptStmtType::BasicStmt)` is removed here - I don't follow exactly how that relates to adding support for `MENU`, and does removing this condition have any impact on how/where these other statements are parsed?

Reply:
I looked at the documentation for [`DIALOG`](https://learn.microsoft.com/en-us/windows/win32/menurc/dialog-resource) and [`DIALOGEX`](https://learn.microsoft.com/en-us/windows/win32/menurc/dialogex-resource) and they support the same "optional statements". I don't know why llvm-rc distinguished between `DIALOG` and `DAILOGEX` here.

I also checked [Resource Hacker](https://angusj.com/resourcehacker/) and it also supports the same "optional statements" for both.

Reply:
Ok, I see - that sounds reasonable. So, were these only supported in `DIALOGEX` before, now they’re supported in both?

It would be good to clarify this in the commit message (PR description), that we’re making the other existing statements available in the other form, too.

Depending on which form was supported before and what gets supported here now, it might be good to add a test to cover that too. But if all the existing statements only were supported in `DIALOGEX` and the new test covers `DIALOG`, that’s probably good enough test coverage.

Reply:
I updated the description. Thanks for the review.

Reply:
Thanks! But does this mean that these statements didn’t work in plain `DIALOG` before? Because looking at `llvm/test/tools/llvm-rc/Inputs/tag-dialog-headers.rc` it seems like these do work in that context already, so I wonder what this changes.

Reply:
Good catch, that was actually a bug I introduced, because `BasicStmt` does not mean `DIALOG`. That's the definition of the `enum`.
```
enum class OptStmtType { BasicStmt, DialogStmt, DialogExStmt };
```

The code already worked for both dialog types. I will fix it.

Reply:
Ok, great, thanks!